src/components/Pin.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+  <div class="pin-container"
+       @mouseover="isHovered = true"
+       @mouseout="isHovered = false"
+       @click="changeImage">
+    <img :src="pinUrl" alt="Pin" :style="{ transform: isHovered ? 'scale(0.9)' : 'scale(1)', transition: 'transform 0.3s ease', width: '75px', height: '100px' }">
```

Comment:
Se puede simplificar esto (imagínate un archivo con muchos componentes y todos así!). 2 ideas que se me ocurren:

* Dejas en style solo el ``transform`` que depende del ``isHovered`` y el resto lo mueves a una clase
* Mueves todo a una clase y utilizas [``v-bind``](https://vuejs.org/api/sfc-css-features.html#v-bind-in-css) con ``isHovered``. Algo así:

```
const scaleCss = computed(() => isHovered ? 'scale(0.9)' : undefined);
...
<style scoped>
.clase {
  transform: v-bind(scaleCss);
  ...
}
....
```

De hecho, se me ocurre también que podemos hacer que la escala sea un ``prop`` con valor por defecto 0.9, de forma que en el padre podamos customizarlo! [Se declaran así](https://vuejs.org/guide/typescript/composition-api.html#props-default-values)